content/projects/gsoc/2021/project-ideas/cloudevents-plugin.adoc

```diff
@@ -0,0 +1,54 @@
+---
+layout: gsocprojectidea
+title: "CloudEvents plugin for Jenkins"
+goal: "Be able to listen to and emit cloudevents from Jenkins."
+category: Plugins
+year: 2021
+status: draft
+sig: docs
+skills:
+- Java
+- Cloudevents sdk
+- Networking
+mentors:
+# To be confirmed
+#- "oleg_nenashev"
+- "waveywaves"
+links:
+  draft: https://docs.google.com/document/d/1xsI6nkEPzXId5npXLrjz3Ydj7jx9Rf8g7SWnTldpeQc/edit#heading=h.f09or9ek462l
+---
+
+=== Background
+As the CI/CD world is moving more towards interoperability between multiple platforms, Jenkins should also be compatible with the same interoperability standards. Some of these standards with respect to communication between different CI/CD platforms are put forth by the cloudevents specification. This spec outlines the structure of cloudevents, which are produced or consumed by entities which support it, hence making those entities compatible with other CI/CD platforms which also support them allowing them to work together. 
+
+Jenkins currently does not support cloudevents, making it hard for users to use it with other platforms which support them. 
+
+==== Project Details
+This project idea proposes to implement a Jenkins plugin which extends the Jenkins to make cloudevents both discoverable and subscribable. Users should be able to set a global configuration to allow users to subscribe and discover cloudevents. 
+
+The project requires the student to start with plugin development from scratch and then work with understanding how to use the Cloudevents Java SDK to create and read events.
+
+In the beginning, the student can work on understanding the Jenkins plugin development by writing code for creating a simple Build Step which creates a CloudEvent, and from there we can move to the GlobalPluginConfiguration on how Jenkins should listen to the cloudevents.
+
+
+=== Links
+There are many examples of such documentation on the web:
+
+* link:https://cloudevents.io/[Cloudevents website]
+* link:https://github.com/cloudevents/sdk-java[Cloudevents Java SDK]
+
+
+=== Quick-start
+
+The project requires the student to start with plugin development from scratch and then work with understanding how to use the Cloudevents Java SDK to create and read events.
+
+In the beginning, the student can work on understanding the Jenkins plugin development by writing code for creating a simple Build Step which creates a CloudEvent, and from there we can move to the GlobalPluginConfiguration on how Jenkins should listen to the cloudevents.
+
+
+=== Newbie-friendly issues
+
```

Comment:
Link to open newbie-friendly issues: https://issues.jenkins.io/browse/JENKINS-64666?jql=status%20%3D%20Open%20AND%20labels%20%3D%20newbie-friendly